### DISCUSSION TITLE
Video with a single remote static image

### DISCUSSION THREAD

**Discussion by guizm0b** [savonet/liquidsoap#2943] [User Support]
Hi,

I have this script below that works fine with a static local images hosted on the server. It also works with remote video urls. Although I would like to use this script to be able to use any static image url from the web. Somehow I can't find how to make this work, or what I'm doing wrong here.
The documentation have examples that uses local single images, but no urls.

Does anyone have an example of this working? 

Thanks!

```
log.level.set(4)

audio_url = "https://example.com/listen"
audio = mksafe(input.http(audio_url))

video = single("https://example.com/image.jpg")

stream = mux_video(video=video, audio)

stream = ffmpeg.encode.audio_video(
    %ffmpeg(
        %audio(codec="aac", b="128k"),
        %video(codec="libx264", b="4000k")
    ),
    stream
)

output_url = "rtsp://rtspserver:8554/mystream"

enc = %ffmpeg(
    format="rtsp",
    rtsp_transport="tcp",
    %audio.copy,
    %video.copy)

output.url(fallible=true, url=output_url, enc, stream)
```

**Comment by vitoyucepi** [comment]
This is indeed a problem that has to be addressed. But I have no simple solution, so could you please create an issue?

I think the file extension resolution is defined here
https://github.com/savonet/liquidsoap/blob/185ac0f1892e4792f640b883b5bd276b5425a5ed/src/libs/protocols.liq#L174-L190

The downloaded file is saved with the `.osb` extension, but it's not recognized [as an extension for image file](https://www.liquidsoap.info/doc-2.1.4/settings.html#file-extensions-used-for-decoding-images-with-ffmpeg).

I'd like to share a hack that will solve the problem. Maybe it'll create an unexpected side effect in future.
```ruby
image_extensions = settings.decoder.file_extensions.ffmpeg.images()
image_extensions = list.add("osb", image_extensions)
settings.decoder.file_extensions.ffmpeg.images.set(image_extensions)
```

**Reply by toots** [reply]
Good point. I've got this coming soon that should make the situation much better: https://github.com/savonet/liquidsoap/pull/2947

**Reply by toots** [reply]
@vitoyucepi I'm adding you as project member so you can start reviewing PRs as well!

**Reply by guizm0b** [reply]
Thank you for the quick response and the hack provided! It works for the time being :)
Is it still necessary to create an issue if this has been addressed in your PR @toots ?

**Reply by toots** [reply]
@guizm0b I don't think so, no. But to get the changes you will have to try the latest `main` or `rolling-release-v2.2.x` once it is merged.
